### PR TITLE
Upgrading chef module using new patterns

### PR DESCRIPTION
**Pull request by marutinandanpandya**

### CODE REVIEW

File: libraries/copperegg_lib.rb [hunk 1]
```diff
@@ -31,64 +31,52 @@ def initialize(apikey, resource_type)
...
-        JSON.parse(json_)
+        JSON.parse(json)
       rescue Exception => e
-        return nil
+        return false
       end
     end
 
     def uninstall_collector()
+      Chef::Log.warn "removing copperegg collector"
```
Comment (gagan93jtg) [inline libraries/copperegg_lib.rb]: Removing Uptime Cloud Monitor collector

File: libraries/copperegg_lib.rb [hunk 2]
```diff
@@ -31,64 +31,52 @@ def initialize(apikey, resource_type)
...
+      return true unless rundir
+
+      mycmd = "/usr/local/revealcloud/revealcloud -V 2>&1 | grep Version"
+      collver = `#{mycmd}`
+      collver = collver.split(' ')[1]
+      collver.chomp!
+
       `curl -sk https://#{@apikey}:U@api.copperegg.com/chef.sh  > /tmp/chef.sh`
-      installer_ver = `grep URL_LINUX_64 /tmp/chef.sh -m 1 | cut -d '/' -f5`
+      installer_ver = `grep URL_LINUX_64 /tmp/chef.sh -m 1 | cutd '/'f5`
```
Comment (gagan93jtg) [inline libraries/copperegg_lib.rb]: is this command available everywhere ? could not find this on my local

File: libraries/copperegg_lib.rb [hunk 3]
```diff
@@ -31,64 +31,52 @@ def initialize(apikey, resource_type)
...
-        `rm -f /tmp/revealcloud_installer.sh`
-        `mv /tmp/chef.sh /tmp/revealcloud_installer.sh`
-        rslt = true
-      elsif (rundir == false) || (confdir == false)
-        rslt = true
+        Chef::Log.warn "Could not get installer version from the API...skipping uninstall"
+        return true
+      end
+
+      if installer_ver==collver
```
Comment (gagan93jtg) [inline libraries/copperegg_lib.rb]: space around operators

File: recipes/_install.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  service 'revealcloud' do
+    supports :restart => true, :status => true, :start => true, :stop => true
+    action [:stop, :start]
+    only_if { File.exists?(agent_config_file) }
+  end
+
+end
+
+unless is_windows
+  if node['copperegg']['create_sshprobe'] && node.attribute?('ec2') && node.attribute.ec2.attribute?('public_hostname')
```
Comment (gagan93jtg) [inline recipes/_install.rb]: did we mention in README about our new SSH probe feature for  the probe which we are creating here ? If not, please update

File: recipes/_linux-collector.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# Recipe:: _linux-collector
+#
+# Copyright 2013-2016 IDERA
+# License:: MIT License
+#
+
+apikey = node['copperegg']['apikey']
+copperegg_url = node['copperegg']['url']
+
+ruby_block "latest_collector_version" do
```
Comment (gagan93jtg) [inline recipes/_linux-collector.rb]: string interpolation

File: recipes/_windows-collector.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#
```
Comment (gagan93jtg) [inline recipes/_windows-collector.rb]: will windows installation work after commenting entire code ?

File: recipes/copperegg-handler.rb [hunk 7]
```diff
@@ -28,11 +28,11 @@
   hostname = hostname + ' (' + node['ec2']['instance_id'] + ')'
 end
 
-chef_handler 'Chef::Handler::Copperegg' do
+chef_handler 'Chef::Handler:'copperegg'' do
```
Comment (gagan93jtg) [inline recipes/copperegg-handler.rb]: single qoutes escape required ?

File: recipes/default.rb [hunk 8]
```diff
@@ -2,152 +2,39 @@
...
+is_windows = (node['platform_family'] == 'windows')
 
-if  node.copperegg.attribute?('tags')
-  cetags = node.copperegg.tags
-end
-
-# If node.copperegg.tags_override exists regardless of value, then do _not_
-# include the chef_environment and chef roles in the tag list
-unless node.copperegg.attribute?('tags_override')
+raise "Add a ['copperegg']['apikey'] attribute to configure this node's Collector." if node['copperegg'] && node['copperegg'][:apikey].nil?
```
Comment (gagan93jtg) [inline recipes/default.rb]: replace [:apikey] with ['apikey']

File: Berksfile [hunk 9]
```diff
@@ -0,0 +1,6 @@
+source "https://api.berkshelf.com"
+
+metadata
+
+
+cookbook 'datadog', '2.0'
```
Comment (gagan93jtg) [inline Berksfile]: why do we need datadog ?

File: CHANGELOG.md [hunk 10]
```diff
@@ -2,6 +2,10 @@
 
 This file is used to list changes made in each version of copperegg cookbook.
 
+## 1.0.0 ()
+
+*
```
Comment (marutinandanpandya) [inline CHANGELOG.md]: once all review and fixes are completed at the time of release

File: attributes/collector.rb [hunk 11]
```diff
@@ -0,0 +1,16 @@
+
+default['copperegg']['must_restart'] = false
+default['copperegg']['update_latest'] = true
+
+if node['platform_family'] == 'windows'
+  default['copperegg']['config_dir'] = "#{ENV['ProgramData']}/Copperegg"
```
Comment (marutinandanpandya) [inline attributes/collector.rb]: ENV['ProgramData'] is C:\Program Files\

File: attributes/collector.rb [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+else
+  default['copperegg']['config_dir'] = '/etc/copperegg'
```
Comment (marutinandanpandya) [inline attributes/collector.rb]: ??

File: recipes/_install.rb (hunk 4 above)
Comment (marutinandanpandya) [inline recipes/_install.rb]: This is not new feature

File: attributes/collector.rb (hunk 11 above)
Comment (gagan93jtg) [inline attributes/collector.rb]: Check https://en.wikipedia.org/wiki/Environment_variable

File: recipes/_install.rb (hunk 4 above)
Comment (gagan93jtg) [inline recipes/_install.rb]: cool

File: recipes/copperegg-handler.rb (hunk 7 above)
Comment (gagan93jtg) [inline recipes/copperegg-handler.rb]: any updates for this ?  @marutinandanpandya

**Pull request merged**